### PR TITLE
Holmanb/phunyguy

### PR DESCRIPTION
**Pull request by phunyguy**
## Proposed Commit Message

```
summary: no more than 70 characters

Made some changes to fix some python 2to3 stuff

Added function to add dns suffix to resolv.conf
```

## Checklist:

 - [ ] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/contributing.html)
 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: cloudinit/distros/gentoo.py [hunk 1]
```diff
@@ -261,5 +267,13 @@ def convert_resolv_conf(settings):
             result += "nameserver %s\n" % ns
     return result
 
+def convert_resolv_conf_search(settings):
+    """Returns a settings string formatted for resolv.conf."""
+    result = ""
+    if isinstance(settings, list):
```
Comment (holmanb) [inline cloudinit/distros/gentoo.py:273]: This function has a single callsite and we know that the argument `settings` is a list, so the if statement will always be true and is therefore a no-op.

File: cloudinit/distros/gentoo.py [hunk 2]
```diff
@@ -261,5 +267,13 @@ def convert_resolv_conf(settings):
...
+        for domain in settings:
+            result += " %s" % domain
```
Comment (holmanb) [inline cloudinit/distros/gentoo.py:275]: Python's `join` is more succinct than looped concatenation. Could we do that instead?

Something like this maybe?
```
def convert_resolv_conf_search(settings):
    """Returns a settings string formatted for resolv.conf."""
    return "domain " + " ".join(settings) + "\n"
```

File: cloudinit/distros/gentoo.py [hunk 3]
```diff
@@ -220,7 +226,7 @@ def set_timezone(self, tz):
         distros.set_etc_timezone(tz=tz, tz_file=self._find_tz_file(tz))
 
     def package_command(self, command, args=None, pkgs=None):
-        cmd = list("emerge")
+        cmd = ["emerge"]
```
Comment (holmanb) [inline cloudinit/distros/gentoo.py:229]: Why this change? Just curious

File: cloudinit/distros/gentoo.py [hunk 4]
```diff
@@ -261,5 +267,13 @@ def convert_resolv_conf(settings):
             result += "nameserver %s\n" % ns
     return result
 
+def convert_resolv_conf_search(settings):
+    """Returns a settings string formatted for resolv.conf."""
+    result = ""
+    if isinstance(settings, list):
+        for domain in settings:
+            result += " %s" % domain
+    return "domain" + result + "\n"
```
Comment (holmanb) [inline cloudinit/distros/gentoo.py:276]: The [docs](https://wiki.gentoo.org/wiki/Resolv.conf#domain) say domain is obsolete. Think we should use `search` instead?

Comment (holmanb) [reply cloudinit/distros/gentoo.py:276]: Just noticed the docs also say "The search option accepts up to six domains". Do you know what the failure mode is? If 7 are provided is the 7th ignored? Or is the config invalid and won't work at all?

Either way we should probably warn if >6.

**Review by holmanb — CHANGES_REQUESTED** [review]
I left a couple comments, some stylistic, some functional.

Let me know what you think :)

File: cloudinit/distros/gentoo.py (hunk 3 above)
Comment (phunyguy) [reply cloudinit/distros/gentoo.py:229]: Otherwise, it was showing up in the logs with an error, because it was trying to run something to the effect of "e m e r g e" as the command, rather than "emerge". I also remember seeing a bug report on launchpad for it with someone else reporting the same problem.

File: cloudinit/distros/gentoo.py (hunk 2 above)
Comment (phunyguy) [reply cloudinit/distros/gentoo.py:275]: I am nowhere close to being an expert, so I'll let you make that call.

File: cloudinit/distros/gentoo.py (hunk 1 above)
Comment (phunyguy) [reply cloudinit/distros/gentoo.py:273]: No idea here.  :)

File: cloudinit/distros/gentoo.py (hunk 4 above)
Comment (phunyguy) [reply cloudinit/distros/gentoo.py:276]: To the questions, I don't know.  Yes I agree, if it will fail with more than 6, then a warning is warranted.

File: cloudinit/distros/gentoo.py (hunk 1 above)
Comment (phunyguy) [reply cloudinit/distros/gentoo.py:273]: Ahh, I fumbled my way through this part.  Guessing the `if` can just be removed then? I wasn't sure there. I copied that code from another part of the script to perform the addition of the search domain(s) IIRC.

File: cloudinit/distros/gentoo.py (hunk 3 above)
Comment (holmanb) [reply cloudinit/distros/gentoo.py:229]: Ouch. Nice catch.

**Comment by holmanb** [comment]
@phunyguy I left some comments on IRC seeing how you wanted to get this merged, but I think you're not in the channel anymore, my questions here in case you wish to comment: https://irclogs.ubuntu.com/2022/08/05/%23cloud-init.html#t22:43

I'm planning to merge this, do a little stylistic cleanup, and then proposing a PR upstream that attributes co-authorship on you. Since you previously said in IRC "I'm going to send you a pr for what I have, for your review, and you can do what you want with it.", I assume this is fine, just wanted to let you know my plans. Got a preferred email address for the co-authored-by line?

**Comment by phunyguy** [comment]
@holmanb sorry, I completely missed your last comment.  I am completely OK with your plans.  I'll pm you the email address in IRC.
